src/worker/start.rs

```diff
@@ -245,47 +271,58 @@ async fn run_task(
 
         stream.send_stream_start().await.map_err(streamer_error)?;
 
+        let stream2 = stream.clone();
+
         let main_fut = async move {
             let stdout = child.stdout.take();
             let stderr = child.stderr.take();
-
             let response = tokio::try_join!(
                 child.wait().map_err(DsError::from),
-                resend_stdio(job_id, job_task_id, 0, stdout, stream.clone())
-                    .map_err(streamer_error),
-                resend_stdio(job_id, job_task_id, 1, stderr, stream.clone())
+                resend_stdio(job_id, job_task_id, 0, stdout, stream2.clone())
                     .map_err(streamer_error),
+                resend_stdio(job_id, job_task_id, 1, stderr, stream2).map_err(streamer_error),
             );
+            status_to_result(response?.0)
+        };
+
+        let guard_fut = async move {
+            let result = tokio::select! {
+                biased;
+                    r = end_receiver => {
+                        Ok(r.unwrap().into())
+                    }
+                    r = main_fut => r
+            };
             stream.close().await.map_err(streamer_error)?;
```

Comment:
Maybe the errors should be combined here somehow? If `result` has a failure and stream closing also fails, the first failure will be lost.

Reply:
Streaming error is an infrastructure error, so it has a higher precedence (and it should almost never happen), so i think that it is ok here

Reply:
Ok, I agree that it should be rare (hopefullly :) ).